stream_alert/sink.py

```diff
@@ -81,6 +84,72 @@ def _get_sns_topic_arn(self):
             topic=topic
         )
 
+    def _chunk_and_encode_sns_messages(self):
+        """Return a list of base64 encoded SNS messages.
+
+        The list of alerts will get divided up across multiple SNS messages to
+        ensure each final, encoded message is less than 256kb in size.
+
+        Returns:
+            List of base64 encoded messages that are sent to be
+            published one at a time.
+        """
+        chunks = []
+        sns_dict = {'default': []}
+
+        for i in range(len(self.alerts)):
+            added = False
+            sns_json_message = self._get_json_from_dict(sns_dict)
+            b64_message_size = self._calculate_b64_size(sns_json_message)
+            new_json_message = ""
+
+            # Check to see if this message size is less than the max sns size
+            # Alerts are at least len('{}'), and therefore this isno need to check <=
+            if b64_message_size < _SNS_MAX_SIZE:
+
+                # Append an alert to the list stored in 'default' key of this dict and
```

Comment:
Hi @strcrzy - I looks like your code will accidentally result in duplicative alerts being sent:
  `sns_dict['default'].append(self.alerts[i])` & `sns_dict = {'default': [self.alerts[i]]}` will send the same alert twice. The two 'len()' checks are also not needed (they're checking values that are already ints representing lens).

I do think you're on the right track though - how does this look:
```            # Check to see if this alert will fit into the current message
            size_with_alert = self._calculate_b64_size(self._get_json_from_dict(self.alerts[i])) + b64_message_size + 2
            if size_with_alert >= _SNS_MAX_SIZE:
                encoded_sns_message = base64.b64encode(sns_json_message)
                chunks.append(encoded_sns_message)
                sns_dict = {'default': [self.alerts[i]]}
                added = True
            else:
                sns_dict['default'].append(self.alerts[i])```